pipeline-modules/app-service/aws/ecr-image-scan.yaml

```diff
@@ -0,0 +1,88 @@
+provisioner: aws
+name: ecr-image-scan
+version: 1
+revision: 1
+displayName: ECR Image Scan
+description: Run ECR Container Scan
+target: ""
+category: scan
+keywords:
+  - scan
+  - sast
+author: CloudCover
+meta: {}
+inputs:
+  properties:
+    working_dir:
+      title: Working directory
+      description: The current working directory to execute the bash script from
+      type: string
+      default: repo
+    allow_critical:
+      title: Allow Critical Vulnerabilities
+      description: Allows builds with critical vulnerabilites
+      type: boolean
+      default: false
+    allow_high:
+      title: Allow High Vulnerabilities
+      description: Allows High vulnerabilites
+      type: boolean
+      default: false
+    allow_warning:
+      title: Allow Warning Vulnerabilities
+      description: Allows Warning vulnerabilites
+      type: boolean
+      default: false
+  internal:
+    - container_repository
+    - container_tag
+template: |
+  version: 0.2
+  env:
+    shell: bash
+  phases:
+    build:
+      commands:
+        - #aws ecr start-image-scan --repository-name={{container_repository}}  --image-id imageTag={{container_tag}}
+        - |
+          while:
+            do 
+              output=$(aws ecr describe-image-scan-findings --repository-name={{container_repository}}  --image-id imageTag={{container_tag}} --output json)
+              status=$(echo $output | jq -r '.imageScanStatus.status')
+              if [ "$status" == "COMPLETE" ]; then
+                echo $output > scan.json
+                break
+              fi
+              sleep 30
```

Comment:
30sec seems too high no? if the result is published on 31st sec, this will end up waiting for 1min.
say if we set it to 10sec, is there any rate limit or cost assiciated?

Reply:
No cost associated or rate limit of this kind that will impact a client. Will change it to 10s.